### DISCUSSION TITLE
Possible to publish to gitlab maven repository

### DISCUSSION THREAD

**Discussion by nazeem-soeltan-maersk** [jreleaser/jreleaser#946] [Q&A]
Hi,

Currently I've managed to create a proof of concept with jReleaser where the artifact(s) & changelog are shown under the Deployments > Releases tab of the gitlab project.

However, our internal maven package repo is hosted on gitlab on [group-level](https://docs.gitlab.com/ee/user/packages/maven_repository/#group-level-maven-endpoint).

Is it possible to also make sure that jReleaser is able to publish it to that specific gitlab package registry? I would've expected an option under https://jreleaser.org/guide/latest/configuration/packagers/index.html. Could you clarify?

**Comment by aalmiray** [comment]
Hi @nazeem-soeltan-maersk as far as I understand publishing to a GitLab Packages using a Maven compatible repository/layout can be performed with regular build tools such as Maven or Gradle. Please see https://docs.gitlab.com/ee/user/packages/maven_repository/

JReleaser supports _uploading_ to generic GitLab packages using the recently released (in [v1.2.0](https://github.com/jreleaser/jreleaser/releases/tag/v1.2.0)) [Gitlab](https://jreleaser.org/guide/latest/configuration/upload/gitlab.html) uploader. This uploader **is not** compatible with Maven layouts, it should only be used to upload binaries such as zips and tars, not jars and poms.

**Reply by nazeem-soeltan-maersk** [reply]
@aalmiray ,

Thanks for the clarification. Is there any proposal to support this in the near future, or is it something that is in contrast with the jReleaser intent?

**Reply by aalmiray** [reply]
Well, syncing to a Maven compatible repository is a feature that has been discussed a couple of times already. We've decided not to provide it for the time being because both Maven and Gradle do a good job about it. However, in the spirit of JReleaser being a "release" tool it makes sense to find a way to support this behavior, we just haven't put it in the roadmap yet.
